src/boot/efi/util.c

```diff
@@ -769,6 +770,114 @@ EFI_STATUS make_file_device_path(EFI_HANDLE device, const char16_t *file, EFI_DE
         return EFI_SUCCESS;
 }
 
+#define QEMU_KERNEL_LOADER_FS_MEDIA_GUID \
+        { 0x1428f772, 0xb64a, 0x441e, {0xb8, 0xc3, 0x9e, 0xbd, 0xd7, 0xf8, 0x93, 0xc7 }}
+
+#define QEMU_BOOT_ORDER_GUID \
+        { 0x668f4529, 0x63d0, 0x4bb5, {0xb6, 0x5d, 0x6f, 0xbb, 0x9d, 0x36, 0xa4, 0x4a }}
+
+/* detect direct kernel boot, i.e. 'qemu -kernel systemd-bootx64.efi' */
+bool is_qemu_direct_kernel(EFI_HANDLE device) {
```

Comment:
I think these two functions should be combined into something like `is_direct_boot`. It's not like you're making a decision based on which kind of direct boot it is (unless that's yet to come?).